src/app/api/datasets/utils.ts

```diff
@@ -0,0 +1,38 @@
+import { pb } from '@/adapters/api'
+
+export async function singleDataset(datasetTitle: string) {
+    const records = await pb
+        .collection('mocDataset')
+        .getFirstListItem(`title="${datasetTitle}"`)
+    return records
+}
+export async function datasetFromUser(userId: string) {
+    async function getDatasetArray(datasetIds: string[]): Promise<any[]> {
+        // pb.collection('mocDataset').getList(1, 50, {
```

Comment:
As we discussed I think it's better to implement this as the single query to fetch all datasets, even though I know this specific part might change in the future we need to know how to do this anyway.

Reply:
Also don't declare the function inside the other function, they can be separate, theres no need to have `getDatasetArray` declared inside `datasetFromUser`